docs/user-guide/github-apps/handling-webhooks.rst

```diff
@@ -0,0 +1,174 @@
+########################
+Handling GitHub webhooks
+########################
+
+A common use for a GitHub App integration in Safir/FastAPI applications is to handle webhooks from GitHub.
+With webhooks, your app can be triggered by events like pushes or pull requests to a repository, and even implement a GitHub Check for pull requests.
+Gidgethub_ provides a routing system for handling webhooks.
+This page shows how to integrate Gidgethub into a Safir/FastAPI application and use the resource models in `safir.github`.
+
+Adding FastAPI handler for GitHub webhooks
+==========================================
+
+The first step is to add a FastAPI handler for the webhook endpoint.
+The URL path for this endpoint corresponds to the webhook callback URL you set up in the GitHub App's settings.
+
+.. code-block:: python
+   :caption: handlers.py
+
+   import httpx
+   from fastapi import APIRouter, Depends, Request, Response, status
+   from gidgethub.sansio import Event
+   from safir.dependencies.logger import logger_dependency
+   from structlog.stdlib import BoundLogger
+
+   external_router = APIRouter()
+   """FastAPI router for all external handlers."""
+
+
+   @external_router.post(
+       "/github/webhook",
+       summary="GitHub App webhook",
+       description=("This endpoint receives webhook events from GitHub"),
+       status_code=status.HTTP_200_OK,
+   )
+   async def post_github_webhook(
+       request: Request,
+       logger: BoundLogger = Depends(logger_dependency),
+   ) -> Response:
+       """Process GitHub webhook events."""
+       webhook_secret = config.github_webhook_secret.get_secret_value()
+       event = Event.from_http(request.headers, body, secret=webhook_secret)
+
+       # Bind the X-GitHub-Delivery header to the logger context; this identifies
+       # the webhook request in GitHub's API and UI for diagnostics
+       logger = logger.bind(github_delivery=event.delivery_id)
+
+       logger.debug("Received GitHub webhook", payload=event.data)
+       # Give GitHub some time to reach internal consistency.
+       await asyncio.sleep(1)
```

Comment:
This seems worth a mention in the documentation. How did you arrive at that delay? How would one know if the delay is too short?

Reply:
I think I got that trick from the Gidgethub docs, who I'm sure got that from experience. I'll mention it in the docs.